### DISCUSSION TITLE
Query about download capabilities

### DISCUSSION THREAD

**Discussion by aero31aero** [CappielloAntonio/tempo#151] [Q&A]
Hi! I'm currently using Symfonium but some of my friends asked me for an app that'd fit almost all those same boxes for them. The biggest one is being able to download a lot of songs but not everything, and do it without manually picking everything. For example, here's a setup I have on Symfonium:

My library: gonic with per language folders, ~10000 tracks.

Rules to download:
- Top 500 most played
- 1000 favorited songs sorted by date last played.
- 500 songs that have never been played, sorted randomly.
- 1000 songs sorted by date added (new songs) for each language.

I'm wondering how much of such a setup could I replicate today in tempo, and if not, is something like this in the plans? Thanks in advance for engaging in this conversation. I'd be happy to help test any beta functionality for this if needed.

**Comment by GallowsDove** [comment]
Hey, thanks for your interest in Tempo.

Tempo doesn't currently support any advanced downloading options. It currently supports downloading specific songs, album or playlists. I plan on adding downloading entire artists, however that is not currently implemented.

I believe that the functionality you've described is fairly nieche, and that there are currently more pressing issues/features to work on.

**Reply by aero31aero** [reply]
In support of such a workflow, I bet there are many people who simply moved to such a setup because their music collection outgrew their phone's storage capacity. So, you try to anticipate what could you probably want to listen to and store a chunk of your library offline, but dynamically, so that if you dislike some music it disappears from the local collection and is replaced by some unheard songs, or some favorited songs, etc.

Either way, I guess my question is answered. I can switch to using beets' smart playlists easily, by making all those filters into regularly updated m3u files.

I'll give this workflow a go, thanks.

**Comment by eddyizm** [comment]
A download feature that I seen in other apps was a checkbox or setting to allow you to download whatever you streamed, eg the random playlist, album, just save a copy while it played. Along with that, there was some limit or storage , say for example like 10GB, after the app reached that amount, it would deleted the oldest files to make way for new stuff.
